nav2_bt_navigator/src/ros_topic_logger.cpp

```diff
@@ -42,11 +42,15 @@ void RosTopicLogger::callback(
 
   // BT timestamps are a duration since the epoch. Need to convert to a time_point
   // before converting to a msg.
-#ifndef _WIN32
+#if defined(__linux__)
```

Comment:
Is there really no way we can represent a timestamp that all of these can use? There are plenty of places in the codebase we use timestamps, but this is the only one with this level of if/elsing on operating systems. If we change `timestamp` to just a `double`, is there no way that all of these can handle it?